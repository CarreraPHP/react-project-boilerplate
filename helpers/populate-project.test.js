const fs = require('fs')
const rimraf = require('rimraf')

const {populateProject} = require('./populate-project')
const {log, error} = require('./logger')

const tmpDir = 'tmp'
const tmpFile = `${tmpDir}/testPopulateProject.md`
const mockMd = `
  sample content
  %%APPNAME%%
  more sample content
  %%HOMEPAGE%%
  and yet more content
`;

describe('populate-project', () => {
  beforeEach(() => {
    // Cleanup the temp
    rimraf.sync('./tmp', {}, () => {
      log('Could not remove tmp directory before test.')
    })
  })
  afterEach(() => {
    // Cleanup the temp
    rimraf.sync('./tmp', {}, () => {
      log('Could not remove tmp directory after test.')
    })
  })

  it('replaces variables in markdown files', async () => {
    // stub new file
    if (!fs.existsSync(tmpDir)){
      fs.mkdirSync(tmpDir);
    }
    fs.writeFileSync(tmpFile, mockMd, (e) => {
      error(e)
      expect(true).toEqual(false) // Force test failure
    })
    // run replacement
    await populateProject({
      root : tmpDir,
      appName : 'mockAppName',
      homepage : 'http://example.com/mockHomepage'
    })
    // load replacement
    const result = fs.readFileSync(tmpFile, 'utf8', (e) => {
      expect(true).toEqual(false) // Force test failure
      error(e)
    })
    expect(result).toMatchSnapshot()
    expect(result.includes('mockAppName'))
    expect(result.includes('http://example.com/mockHomepage'))
  })
})
