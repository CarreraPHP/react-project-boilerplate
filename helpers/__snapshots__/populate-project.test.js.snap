// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`populate-project replaces variables in markdown files 1`] = `
"
  sample content
  mockAppName
  more sample content
  http://example.com/mockHomepage
  and yet more content
"
`;
